builtin/providers/azurerm/loadbalancer.go

```diff
@@ -0,0 +1,54 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/terraform/helper/resource"
+	"net/http"
+)
+
+func retrieveLoadbalancerById(loadBalancerId string, meta interface{}) (*network.LoadBalancer, bool, error) {
+	loadBalancerClient := meta.(*ArmClient).loadBalancerClient
+
+	id, err := parseAzureResourceID(loadBalancerId)
+	if err != nil {
+		return nil, false, err
+	}
+	name := id.Path["loadBalancers"]
+	resGroup := id.ResourceGroup
+
+	resp, err := loadBalancerClient.Get(resGroup, name, "")
+	if err != nil {
+		return nil, false, fmt.Errorf("Error making Read request on Azure Loadbalancer %s: %s", name, err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
See my previous comment about ArmClients and 404 responses raising errors. I suspect this will fail a `_disappears()` acceptance test if there was one present.
Moving the conditional inside the `if err != nil` check should work as intended.